app/scripts/modules/amazon/src/loadBalancer/configure/application/TargetGroups.tsx

```diff
@@ -369,27 +352,42 @@ export class TargetGroups extends React.Component<ITargetGroupsProps, ITargetGro
                           </span>
                           <span className="wizard-pod-content">
                             <label>Timeout </label>
-                            <SpInput
-                              className="form-control input-sm inline-number"
-                              error={tgErrors.healthCheckTimeout}
-                              name="healthCheckTimeout"
-                              required={true}
-                              value={targetGroup.healthCheckTimeout}
-                              onChange={event =>
-                                this.targetGroupFieldChanged(index, 'healthCheckTimeout', event.target.value)
-                              }
-                            />
+                            {(has6sTimeout || has10sTimeout) && <HelpField id="aws.targetGroup.healthCheckTimeout" />}
+                            {has6sTimeout || has10sTimeout ? (
+                              <SpInput
+                                disabled={true}
+                                className="form-control input-sm inline-number"
+                                error={tgErrors.healthCheckTimeout}
+                                name="healthCheckTimeout"
+                                required={true}
+                                value={has6sTimeout ? 6 : 10}
+                                onChange={event =>
+                                  this.targetGroupFieldChanged(index, 'healthCheckTimeout', event.target.value)
+                                }
+                              />
+                            ) : (
+                              <SpelNumberInput
+                                error={tgErrors.healthCheckTimeout}
+                                required={true}
+                                value={targetGroup.healthCheckTimeout}
+                                min={2}
+                                max={120}
+                                onChange={(value: string) =>
```

Comment:
I noticed that everywhere you're using `SpelNumberInput` (for both ALB/NLB) the type of this callback has been narrowed to `string` — what's the context behind that? Is there something downstream of these callbacks that gets angry if you let the `number | string` type flow through?

I only ask because in theory (not always in practice) it's preferable to let the types that are already defined flow, and in this case it seems like the narrowed `string` type also isn't entirely accurate re: what will actually happen at runtime?